### PR TITLE
Add bob recipe

### PR DESCRIPTION
**Pull request by 183amir**

### CODE REVIEW

**Comment by 183amir** [comment]
Too soon for this.

**Comment by 183amir** [comment]
The requirements are pinned in bob: https://github.com/idiap/bob/blob/master/requirements.txt

File: recipes/bob/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    - bob.sp ==2.0.4
+
+test:
+
+  imports:
+    - bob
+    - bob.script
+    - bob.test
+  commands:
+    - nosetests -sv bob --exclude=test_cflags_other --exclude=test_extensions
```
Comment (183amir) [inline recipes/bob/meta.yaml]: The tests are excluded as in: https://github.com/conda-forge/bob.extension-feedstock/blob/master/recipe/meta.yaml#L45

**Comment by 183amir** [comment]
I would appreciate it if you could review this today.

File: recipes/bob/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    - python
+    - bob.ap ==2.0.4
+    - bob.blitz ==2.0.8
+    - bob.core ==2.1.2
+    - bob.db.atnt ==2.0.3
+    - bob.db.base ==2.0.6
+    - bob.db.iris ==2.0.4
+    - bob.db.mnist ==2.0.3
+    - bob.db.verification.utils ==2.0.7
+    - bob.db.wine ==2.0.3
```
Comment (msarahan) [inline recipes/bob/meta.yaml]: all of these static pins make me very nervous about the future of this package.  Is there a reason you need to pin with ==?  Would >= be OK?

When you pin like this, the chances for unsatisfiable dependencies increase.

File: recipes/bob/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    - bob.learn.em ==2.0.8
+    - bob.learn.libsvm ==2.0.3
+    - bob.learn.linear ==2.0.7
+    - bob.learn.mlp ==2.0.11
+    - bob.math ==2.0.3
+    - bob.measure ==2.1.1
+    - bob.sp ==2.0.4
+
+test:
+
```
Comment (msarahan) [inline recipes/bob/meta.yaml]: extra newline here

**Comment by msarahan** [comment]
Please reconsider the pinning.  Otherwise, +1 from me.

File: recipes/bob/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/bob/meta.yaml]: Agreed. This seems like a pain to maintain like this.

Comment (183amir) [inline recipes/bob/meta.yaml]: As I mentioned, these requirements are in `requirements.txt` of the package. They want everything to be pinned for each release.

Comment (183amir) [inline recipes/bob/meta.yaml]: The Pypi package `bob` is like a meta package. It pins 31 other core `bob.*` packages on each release. The `bob` package itself does not add any new functionality so you can not install this one and install any version of `bob.*` packages that you want.

Comment (msarahan) [inline recipes/bob/meta.yaml]: OK, fair enough.  Fine with me.

**Comment by msarahan** [comment]
apparently circle CI wasn't paying attention to [skip ci].  Merging.

**Comment by 183amir** [comment]
Thanks @msarahan

**Comment by jakirkham** [comment]
> apparently circle CI wasn't paying attention to [skip ci].  Merging.

Yeah, I think CircleCI gets messed up with merge commits from PRs and therefore ignore `[skip ci]` in commits. If you add it to the PR title, it will pick it up, but that isn't what anyone should be doing.

**Comment by 183amir** [comment]
@conda-forge/core Now that I am finished with packaging `bob` packages, I was wondering if it is possible that I become a member of `conda-forge` (at least with push access in `staged-recipes`).
I have gained some experience during this process and would be happy to spend my free time helping conda-forge community.

**Comment by msarahan** [comment]
Not sure what's necessary to carry this out, but I'm in favor of it.  Your offer of help is much appreciated.

**Comment by jakirkham** [comment]
Absolutely, I think you have become quite proficient at this @183amir and we would certainly appreciate someone else helping out in the review process.

**Comment by jakirkham** [comment]
cc @ocefpaf

**Comment by jakirkham** [comment]
cc @pelson

**Comment by jakirkham** [comment]
I went ahead and did it (hope that's not controversial). You now have write privileges, @183amir.

**Comment by 183amir** [comment]
@jakirkham and @msarahan thank you very much for your support.

**Comment by ocefpaf** [comment]
I was going to add my :+1: and add the request to the meeting agenda, but that :ship: has :boat: 

Welcome aboard @183amir! It would be nice to have you in our next meeting.

**Comment by jakirkham** [comment]
> I was going to add my :+1: and add the request to the meeting agenda, but that :ship: has :boat:

Oops. 😳  Sorry that was controversial.

**Comment by ocefpaf** [comment]
> Oops. :flushed: Sorry that was controversial.


Not really.  I just like to leave things breath a little bit :wink:

**Comment by 183amir** [comment]
> Welcome aboard @183amir! It would be nice to have you in our next meeting.

@ocefpaf Thank you and please make sure to ping me for the meetings.

**Comment by jakirkham** [comment]
Meeting are all on the [hackpad]( https://conda-forge.hackpad.com/ ). Every two weeks on Friday at 1400UTC.

**Comment by ocefpaf** [comment]
> @ocefpaf Thank you and please make sure to ping me for the meetings.

@jakirkham beat me to it (again :wink:)

@183amir keep your eye on the hackpad. The meetings are open and you don't need to be a core member to participate.

PS: we also have an unfinished Code of Conduct doc that needs some love from the core members:

https://docs.google.com/document/d/10dxX0Zse0Rx1HqsxC73Wfsghmy5m8PP8cHuBIOhWKpc/edit

**Pull request merged**